### ISSUE TITLE
Implement support for library schemas

### ISSUE THREAD

**Issue by happy5214** [hed-standard/hed-javascript#67]
The HED 3 spec allows people to implement library schemas to supplement the base schema. We must support these library schemas by writing code to load them, modifying the parser to detect when they are used in tags, and adjusting existing code to reference their `SchemaEntry` data when tags are pulled from them.

**Comment by garrettmflynn** [comment]
Is there anything I could do to help get this feature implemented and added to the BIDS Validator? I'm working with Jorge Valdes-Sosa and Jorge Bosch from the Global Brain Consortium to develop [an application to read/write HED SCORE tags to BIDS datasets](https://garrettflynn.com/gen/), and a major concern of ours is that we currently have to ignore the HED errors generated during validation. 

Looks like there's an open pull request (#75), but I thought I'd offer my support if it's useful.

**Comment by VisLab** [comment]
Yes this is a top priority for us too. There are three aspects to getting that done:
1.  Changing the BIDS specification.
2.  Modifying the calling stub in the `bids-validator` to construct the appropriate schema object to pass to the `hed-validator`.
3.  Modifying the `hed-validator` to validate the HED annotations that use library schema.

Here is the status of these things:

**The BIDS spec has to be changed.** There is a [Draft PR](https://github.com/bids-standard/bids-specification/pull/1106) on the BIDS specification. However, I am holding off on moving this to a PR, because I think a couple of changes need to be made.  Hopefully, we'll resolve those soon and finalize the interface in `dataset_description.json`.

**Modifying the stub in the bids-validator**:  Once the spec has been agreed upon, then work can begin on this.

**Modifying the `hed-validator` to validate with library schema**:  @happy5214 is working on this part.

Bottom line:  I am working to resolve No. 1 and @happy5214 is working on No. 3. 

Once No. 1 has been resolved and moved to a regular pull request in the BIDS Specification, we could use your help on No. 2. 

You would need to look at how the `bids-validator` interacts with the `hed-validator`.  They used to be tightly coupled, but we have worked hard to separate them with a well-defined interface so that they can be updated independently.  This has resulted in a better structure on both sides and better error reporting on the HED side.  Note: @dungscout96 @dorahermes @tpatpa @smakeig @sappelhoff @Remi-Gau @effigies @rwblair.

**Comment by garrettmflynn** [comment]
Perfect! I'll keep my eye on that Draft PR and coordinate further once it's through.

**Comment by garrettmflynn** [comment]
Hi @VisLab and @happy5214! What's the status of each of the three points mentioned above?  

I see that there's a draft pull request—though it has not been merged to close out this issue.

**Comment by VisLab** [comment]
We are working very hard to get this working.  All commits are on the
library-schemas branch.   Our goal is to get everything in for the next
release of the BIDS specification.

**Comment by garrettmflynn** [comment]
Perfect. Thank you!
